### ISSUE TITLE
confusion about losses.py

### ISSUE THREAD

**Issue by huangcong77** [fizyr/keras-retinanet#191]
Hi
sorry to bother you again.
I want make a little change to focal loss.
```
def focal(alpha=0.25, gamma=2.0):
    def _focal(y_true, y_pred):
```
But I don't know `y_true.shape`, `y_pred.shape` and what each dimension of `y_true` and `y_pred` means.
I could appreciate it  if you could solve my doubts

**Comment by hgaiser** [comment]
`y_true` are the true labels, `y_pred` the predicted labels.

Both of these blobs are "one-hot" vectors. Meaning their shape is as follows:

`(None, num_anchors, num_classes)` where `num_anchors` are the number of anchors for a given image and `num_classes` the number of classes. Each anchors has therefore `num_classes` values assigned to it.

For `y_true` these values are all -1 if the anchor should be ignored, all zeros if it is a background anchor and all zeros with one 1 on the index of the class if it is a foreground anchor.

For `y_pred` it is the sigmoid activation of the classification subnet.

Does this answer your questions?

**Comment by awilliamson** [comment]
In addition to what @hgaiser has posted. Keras has some documentation regarding making your own loss functions, and what default arguments are passed to this function, this might clear up some confusion. https://keras.io/losses/#usage-of-loss-functions

**Comment by huangcong77** [comment]
Thank you so much！I got it!
